lib/spack/llnl/util/filesystem.py

```diff
@@ -2046,15 +2054,23 @@ def find_libraries(libraries, root, shared=True, recursive=False):
         message = message.format(find_libraries.__name__, type(libraries))
         raise TypeError(message)
 
+    if is_windows:
+        static = "lib"
+        shared = "dll"
+    else:
+        # Used on both Linux and macOS
+        static = "a"
+        shared = "so"
```

Comment:
@mwkrentel @scheibelp @johnwparent That's the error. Here the `shared` argument is overridden, so we'll always hit the `True` condition in the conditional below.

Reply:
Oh just found my way here looking into the `armadillo` build failure in the tutorial stack from today.  It can't find libraries in the relocated `superlu` binary.  I guess this is the source of the problem?

Reply:
@zackgalbreath I think this :point_up: could the problem we've been looking into.

Reply:
@alalazo Noticed this a bit ago, I have a solution for this if you haven't started.

Reply:
No I didn't work on a solution, just getting to the point of figuring out the cause.